From 323573a2d36fa746b60f2ea0524e6f42a36cc9f3 Mon Sep 17 00:00:00 2001
From: Arunpandi Kannan <quic_arunpand@quicinc.com>
Date: Fri, 9 Feb 2024 14:31:24 +0530
Subject: [PATCH] hostapd: remove crypto_bignum_mod computation in pkex key
 derive.

Remove crypto_bignum_mod computation change while deriving
PKEX Q key in PKEX exchange request since it is causing the issue
in Pkex exchange.
crypto_bignum_mod added mbedtls crypto encryption but hostapd
uses openssl crypto encryption.

Signed-off-by: Arunpandi Kannan <quic_arunpand@quicinc.com>
---
 src/common/dpp_crypto.c | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

--- a/src/common/dpp_crypto.c
+++ b/src/common/dpp_crypto.c
@@ -1588,9 +1588,7 @@ dpp_pkex_derive_Qr(const struct dpp_curv
 	Pr = crypto_ec_key_get_public_key(Pr_key);
 	Qr = crypto_ec_point_init(ec);
 	hash_bn = crypto_bignum_init_set(hash, curve->hash_len);
-	if (!Pr || !Qr || !hash_bn ||
-	    crypto_bignum_mod(hash_bn, crypto_ec_get_prime(ec), hash_bn) ||
-	    crypto_ec_point_mul(ec, Pr, hash_bn, Qr))
+	if (!Pr || !Qr || !hash_bn || crypto_ec_point_mul(ec, Pr, hash_bn, Qr))
 		goto fail;
 
 	if (crypto_ec_point_is_at_infinity(ec, Qr)) {
